dsalgo/sort.py

```diff
@@ -18,169 +18,205 @@ def __new__(self, array, algo, reverse=False):
-    we combine the results by creating a sorted array A[p..r] from two sorted subarrays A[p..q] and A[q+1, r].
+    When the conquer step reaches the base step and
+    we get two sorted subarrays A[p..q] and A[q+1, r] for array A[p..r],
+    we combine the results by creating a sorted array A[p..r]
+    from two sorted subarrays A[p..q] and A[q+1, r].
+     args:
+      :array:(list) -> list to be sorted
+      :reverse:(boolean) -> default = False,
+                            can be True for sort
+                            in reverse order
     """
-    
-    if len(array) >1: 
-        mid = len(array)//2  # mid
-        left = array[:mid]   # Dividing the array elements  
-        right = array[mid:]  # into 2 halves 
-  
-        merge(left)        # Sorting the first half 
-        merge(right)       # Sorting the second half 
+
+    if len(array) > 1:
+        mid = len(array) // 2  # mid
+        left = array[:mid]  # Dividing the array elements
+        right = array[mid:]  # into 2 halves
+
+        merge(left)  # Sorting the first half
+        merge(right)  # Sorting the second half
         i = j = k = 0
-          
-        # Copy data to left[] and right[] 
-        while i < len(left) and j < len(right): 
-            if left[i] < right[j]: 
-                array[k] = left[i] 
-                i+= 1
-            else: 
-                array[k] = right[j] 
-                j+= 1
-            k+= 1
-          
-        # Checking if any element was left 
-        while i < len(left): 
-            array[k] = left[i] 
-            i+= 1
-            k+= 1
-          
-        while j < len(right): 
-            array[k] = right[j] 
-            j+= 1
-            k+= 1
-    if reverse==True :
-     return array[::-1]
+
+        # Copy data to left[] and right[]
+        while i < len(left) and j < len(right):
+            if left[i] < right[j]:
+                array[k] = left[i]
+                i += 1
+            else:
+                array[k] = right[j]
+                j += 1
+            k += 1
+
+        # Checking if any element was left
+        while i < len(left):
+            array[k] = left[i]
+            i += 1
+            k += 1
+
+        while j < len(right):
+            array[k] = right[j]
+            j += 1
+            k += 1
+    if reverse:
+        return array[::-1]
     return array
 
-def bubble_recursion(array,reverse=False): 
-    for i, num in enumerate(array): 
-        try: 
-            if array[i+1] < num: 
-                array[i] = array[i+1] 
-                array[i+1] = num 
-                bubble_recursion(array) 
-        except IndexError: 
+
+def bubble_recursion(array, reverse=False):
+    for i, num in enumerate(array):
+        try:
+            if array[i + 1] < num:
+                array[i] = array[i + 1]
+                array[i + 1] = num
+                bubble_recursion(array)
+        except IndexError:
```

Comment:
You need to cover this error in the test case too

Reply:
> You need to cover this error in the test case too

If it's got index error the tests already show that one and print that error on the terminal while testing, So I think we no need to cover that separately.

Reply:
No, it prints nothing on the terminal.
You can skip this by removing the try & except and changing the loop to iterate over n-1 elements.
Also, try-except also works fine.
There is no problem with that by me.

Reply:
Added details for exception